### ISSUE TITLE
dump.exfat should do more than just disk/volume information

### ISSUE THREAD

**Issue by hgkamath** [exfatprogs/exfatprogs#173]
The advantages of exfat filesystem are that (A) it is simple, (B) has huge maxsize limit and (C) has a bit for indicating contiguous-ness allowing for read optimization.
The simplicity is a trade off against not having a journal and disk failure tolerance, robustness.

A use case is to store large static container files like virtual-disk images, that are seldom resized/moved. They may also be defragmented in the case of hdd (not ssd) as contiguous files.
More importantly, knowing where the used file blocks are helps in case of disaster recovery.

-s show only information from superblock(s)
-u show comma separated (or line separated) offset ranges of blocks/clusters of blocks/clusters used by any file
-f <filepath> should show comma separated (or line separated) offset ranges of blocks/clusters associated with a file
-R <dirpath> should recursively descend a directory show comma separated (or line separated) offset ranges of blocks/clusters associated with contained files  (this functionality may also be simulated by using a shell script to loop over output of find command and use -f argument on each file)

Such information may help during disaster recovery.

**Comment by namjaejeon** [comment]
Okay, I can not understand fully what you do. Can you implement it and show the execution log and usage at real ?

**Comment by hgkamath** [comment]
Essentially, dump tool for kernel-driver based exfatprogs should give information similar to the fuse-driver based exfat-utils
compare
https://manpages.debian.org/experimental/exfatprogs/dump.exfat.8.en.html 
https://manpages.debian.org/unstable/exfat-utils/dumpexfat.8.en.html

**Comment by namjaejeon** [comment]
fsck is ready on exfat-next branch. Please check whether it is covered for your request.

git clone --branch=exfat-next https://github.com/exfatprogs/exfatprogs

**Comment by hgkamath** [comment]
Not really, the issue request was regarding being able to do some get file-fragment information, in case of need to do manual-scraping/recovery before even attempting fs-repair. Online fs-repair tools modify on-disk data, sometimes as a user one can only say "Y" to whatever is wants confirmation on. Pre-saved FAT file-fragment info dumps is one contingency option. This option could be useful for an ExFAT volume which has a few large rarely size changing virtual-disks.

Nevertheless it is good to know exfat fsck is ready.    Good-work !

As of now, my filesystems/hdds are fine, and I don't have any corruption to test against, and I hope they remain that way.
```
[root@fedora gana]# rpm -qa | grep exfat
exfatprogs-1.1.3-2.fc36.x86_64
```
koji builds for fc37 are at exfatprogs-1.1.3.
https://koji.fedoraproject.org/koji/packageinfo?packageID=32895

**Comment by namjaejeon** [comment]
@hgkamath We have released exfat 1.2.0 version. Please check it.

**Comment by namjaejeon** [comment]
@hgkamath Ping?

**Comment by hgkamath** [comment]
switched back to linux now, am checking now
